packages/astro/src/content/vite-plugin-content-imports.ts

```diff
@@ -65,26 +66,30 @@ export function astroContentImportPlugin({
 	const plugins: Plugin[] = [
 		{
 			name: 'astro:content-imports',
-			async load(viteId) {
-				if (isContentFlagImport(viteId, contentEntryExts)) {
-					const { fileId } = getFileInfo(viteId, settings.config);
+			async transform(_, viteId) {
+				if (isContentFlagImport(viteId)) {
+					const fileId = viteId.split('?')[0];
 					const { id, slug, collection, body, data, _internal } = await setContentEntryModuleCache({
 						fileId,
 						pluginContext: this,
 					});
 
 					const code = escapeViteEnvReferences(`
-export const id = ${JSON.stringify(id)};
-export const collection = ${JSON.stringify(collection)};
-export const slug = ${JSON.stringify(slug)};
-export const body = ${JSON.stringify(body)};
-export const data = ${devalue.uneval(data) /* TODO: reuse astro props serializer */};
-export const _internal = {
-	filePath: ${JSON.stringify(_internal.filePath)},
-	rawData: ${JSON.stringify(_internal.rawData)},
-};
-`);
-					return { code };
+						export const id = ${JSON.stringify(id)};
+						export const collection = ${JSON.stringify(collection)};
+						export const slug = ${JSON.stringify(slug)};
+						export const body = ${JSON.stringify(body)};
+						export const data = ${devalue.uneval(data) /* TODO: reuse astro props serializer */};
+						export const _internal = {
+							filePath: ${JSON.stringify(_internal.filePath)},
+							rawData: ${JSON.stringify(_internal.rawData)},
+						};`);
+
+					if (settings.config.vite.build?.sourcemap) {
+						const s = new MagicString(code);
+						return { code: s.toString(), map: s.generateMap() };
+					}
```

Comment:
I believe for virtual files or generated content, we can return `map: { mappings: '' }` to note that there's no sourcemap (it's the file itself). This might be faster than using `MagicString`.

https://rollupjs.org/plugin-development/#source-code-transformations

Reply:
Sounds good to me! It's virtual anyways so I'm not sure what use a sourcemap would have